fmriprep/workflows/util.py

```diff
@@ -17,12 +17,35 @@
 from niworkflows.nipype.pipeline import engine as pe
 from niworkflows.nipype.interfaces import utility as niu
 from niworkflows.nipype.interfaces import fsl, afni, ants, freesurfer as fs
-from niworkflows.interfaces.registration import FLIRTRPT, BBRegisterRPT
+from niworkflows.interfaces.registration import FLIRTRPT, BBRegisterRPT, MRICoregRPT
 from niworkflows.interfaces.masks import SimpleShowMaskRPT
 
 from ..interfaces.images import extract_wm
 
 
+def compare_xforms(test_mat, fallback_mat):
+    import numpy as np
+    from transforms3d.affines import decompose44
+    from transforms3d.axangles import mat2axangle
+    # [[R], [A], [S], [1]] = mat1.dot([[x], [y], [z], [1]])
+    mat1 = np.loadtxt(fallback_mat)
+    # [[R'], [A'], [S'], [1]] = mat2.dot([[x], [y], [z], [1]])
+    mat2 = np.loadtxt(test_mat)
+    # [[R'], [A'], [S'], [1]] = mat2.dot(mat1_inv.dot([[R], [A], [S], [1]]))
+    comp = mat2.dot(np.linalg.pinv(mat1))
+    trans, rotation_matrix, scales, shears = decompose44(comp)
+
+    shift_thresh = 5         # mm
+    rot_thresh = np.pi / 36  # 5 degrees
+    scale_thresh = 1.1       # scale factor
+
+    shift_magnitude = np.sqrt(trans.dot(trans))  # 2-norm
+    rot = mat2axangle(rotation_matrix)[1]
+    max_scale = np.max(np.abs(scales))
```

Comment:
Two changes here:

1) Use the 2-norm to get total shift, not just max in any direction
2) Increase shift threshold to 5mm